lib/national_identification_number/swedish.rb

```diff
@@ -25,8 +25,22 @@ def self.generate(date=nil, serial=nil)
 
     protected
 
+    def parse_twelve_digits
+      return unless @number.match(/\A(\d{4})(\d{2})(\d{2})(\d{4})\Z/)
```

Comment:
Perhaps use named captures instead of $1 etc?

Reply:
Yes, many things would benefit from a refactor here. But given the state of this project as a whole I don't think it's worth it.

Reply:
I've merged it because I already pushed a new gem version to rubygems and doing changes I would need to release again.